packages/react-aria-components/test/Autocomplete.test.tsx

```diff
@@ -98,29 +100,77 @@ let ListBoxWithSections = (props) => (
+  let {contains} = useFilter({sensitivity: 'base'});
+  let filter = (textValue, inputValue) => contains(textValue, inputValue);
 
-let ControlledAutocomplete = ({autocompleteProps = {}, inputProps = {}, children}: {autocompleteProps?: any, inputProps?: any, collectionProps?: any, children?: ReactNode}) => {
+  return (
+    <UNSTABLE_Autocomplete filter={filter} {...autocompleteProps}>
+      <SearchField {...inputProps}>
+        <Label style={{display: 'block'}}>Test</Label>
+        <Input />
+        <Text style={{display: 'block'}} slot="description">Please select an option below.</Text>
+      </SearchField>
+      {children}
+    </UNSTABLE_Autocomplete>
+  );
+};
+
+let ControlledAutocomplete = ({autocompleteProps = {}, inputProps = {}, children}: {autocompleteProps?: any, inputProps?: any, children?: ReactNode}) => {
   let [inputValue, setInputValue] = React.useState('');
+  let {contains} = useFilter({sensitivity: 'base'});
+  let filter = (textValue, inputValue) => contains(textValue, inputValue);
 
   return (
-    <Autocomplete inputValue={inputValue} onInputChange={setInputValue} {...autocompleteProps}>
+    <UNSTABLE_Autocomplete inputValue={inputValue} onInputChange={setInputValue} filter={filter} {...autocompleteProps}>
       <SearchField {...inputProps}>
         <Label style={{display: 'block'}}>Test</Label>
         <Input />
         <Text style={{display: 'block'}} slot="description">Please select an option below.</Text>
       </SearchField>
       {children}
-    </Autocomplete>
+    </UNSTABLE_Autocomplete>
+  );
+};
+
+let AsyncFiltering = ({autocompleteProps = {}, inputProps = {}}: {autocompleteProps?: any, inputProps?: any, children?: ReactNode}) => {
+  let list = useAsyncList<AutocompleteItem>({
+    async load({filterText}) {
+      let json = await new Promise(resolve => {
+        setTimeout(() => {
+          resolve(filterText ? items.filter(item => {
+            let name = item.name.toLowerCase();
+            for (let filterChar of filterText.toLowerCase()) {
+              if (!name.includes(filterChar)) {
+                return false;
+              }
+              name = name.replace(filterChar, '');
+            }
+            return true;
+          }) : items);
+        }, 300);
+      }) as AutocompleteItem[];
+
+      return {
+        items: json
+      };
+    }
+  });
+
+  return (
+    <UNSTABLE_Autocomplete inputValue={list.filterText} onInputChange={list.setFilterText} {...autocompleteProps}>
+      <SearchField {...inputProps}>
+        <Label style={{display: 'block'}}>Test</Label>
+        <Input />
+        <Text style={{display: 'block'}} slot="description">Please select an option below.</Text>
+      </SearchField>
+      <Menu
+        items={list.items}
+        onAction={onAction}
+        onSelectionChange={onSelectionChange}>
+        {item => <MenuItem id={item.id}>{item.name}</MenuItem>}
+      </Menu>
```

Comment:
We may want to look into giving Menu/Listbox a loading indicator component like Table has. Menu itself doesn't even have the notion of an empty state

Reply:
yeah idk if the loader should be on the menu or inside the searchfield like ComboBox/SearchAutocomplete. We might need to talk to design about that one

Reply:
I'll make a note to ask them